community/planning/batch_queuer_strategies.md

```diff
@@ -0,0 +1,395 @@
+* Explicit user-defined dependencies
+* Optional implicit dependencies based on order
+* Resubmission
+* Service allocation
+* CAP theorem
+* Batch status resolution
+
+### Underlying DLT characteristics
+
+* Guaranteed vs. non-guaranteed edit order
+* Persistence
+
+#### DLT types
+
+For the purposes of the queuer strategies, there are two types of DLTs:
+
+* __DLTs with guaranteed order__ - These DLTs ensure that the order in which
+  edits to state are submitted is the same order in which they will be applied.
+  These DLTs use consensus algorithms, such as two-phase commit, that do not
+  rely on gossip protocol.
+* __DLTs without guaranteed order__ - These DLTs rely on broadcast algorithms
+  (frequently the gossip protocol) that cannot guarantee the order in which
+  edits are applied
+
+Differentiating between the two is important because they imply different
+performance opportunities given that Grid must guarantee edit order.
+
+#### Edit order
+
+Edit order is simply the order in which edits are applied to the same part of
+the ledger state.
+
+Say we have a purchase order (PO) contract in Grid, and the price for the good
+is driven by changes in an underlying commodity’s spot price. Let’s also say
+the current price listed on the PO is 2.75 USD. If we update the PO price to
+3.00 USD and shortly thereafter need to update the PO price again to 3.25 USD
+due to a spike in the commodity price, our edit order would for that PO’s price
+would be 2.75 → 3.00 → 3.25.
+
+However, without a guarantee of edit order, the edits could be applied
+2.75 → 3.25 → 3.00, leaving the final PO contract price at 3.00 USD. Clearly,
+this could be disastrous.
+
+#### Guaranteed order
+
+For strategies with guaranteed order, our strategy considerations are contained
+to its implementation inside Grid itself. So, our goal is to optimize a
+strategy’s performance (i.e. throughput) given certain scenarios or constraints
+(ex. high submitter-to-service ratios or resubmission timers, respectively). We
+simply submit batches as fast as we can, slowing the pace if we get back
+pressure from the DLT. These strategy implementations are simpler than those
+that handle DLTs without order guarantees.
+
+#### Non-guaranteed order
+
+For strategies to handle DLTs without order guarantees, we must ensure the
+strategies themselves provide the order guarantee; this becomes our top design
+priority. Performance then becomes the second priority.
+
+These strategies are necessarily more complex and less performant than the
+guaranteed-order strategies. Improvements in performance usually require
+significant increases in complexity.
+
+#### Persistence
+
+In developing queuing strategies, we want to also consider how DLTs persist
+their pending batch queues, i.e. how batches are held prior to being committed.
+In the normal course of operation, this will not matter to the queuer. However,
+were the DLT node to go restart, this can impact a queuing strategy's
```

Comment:
I would change this to "were the DLT node to restart..."

Reply:
ha, yes, thanks